nemo/core/neural_types/nmtensor_registry.py

```diff
@@ -0,0 +1,88 @@
+# =============================================================================
+# Copyright (c) 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+
+class NmTensorNameRegistry:
```

Comment:
1) I really do not understand why do we need this entity - each neural graph already records ALL the tensors in it, an through AppState you have access to all graphs (AppState().graphs -> dict of graphs)
2) There is ObjectRegistry class which a) creates weak references to objects and b) generates unique names
3) I really do not understand why do you need ALL Tensors, where ALL == tensors from all graphs. This has serious implications. I honestly think that a given callback should operate within a scope of a single graph only.

Reply:
1. I need a mapping from a user-defined user to a tensor.unique_name
2. As I have mentioned in your NeuralGraph PR, tensors already have a property called `unique_name` that guarantee uniqueness so there is no reason for ObjectRegistry
3. Well as we have not moved forward with NeuralGraphs inside of actions.py, there is no "currently running graph" that I can fetch tensors from.

Reply:
Thanks for the reply. Closing.